File: app/src/main/java/com/gersonsilvafilho/petfunding/detail/ImagePagerAdapter.java
package com.gersonsilvafilho.petfunding.detail;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.gersonsilvafilho.petfunding.R;

class ImagePagerAdapter extends PagerAdapter {

    int[] mResources = {
            R.drawable.nino,
            R.drawable.nino2,
            R.drawable.nino3,
            R.drawable.nino4,
    };

    Context mContext;
    LayoutInflater mLayoutInflater;
 
    public ImagePagerAdapter(Context context) {
        mContext = context;
        mLayoutInflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
    }
 
    @Override
    public int getCount() {
        return mResources.length;
    }
 
    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == ((LinearLayout) object);
    }
 
    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View itemView = mLayoutInflater.inflate(R.layout.image_pager, container, false);
 
        ImageView imageView = (ImageView) itemView.findViewById(R.id.imageView);
        imageView.setImageResource(mResources[position]);
 
        container.addView(itemView);
 
        return itemView;
    }
 
    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        container.removeView((LinearLayout) object);
    }
}

File: app/src/main/java/com/gersonsilvafilho/petfunding/detail/ImagePagerAdapter.kt
package com.gersonsilvafilho.petfunding.detail

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.LinearLayout

import com.gersonsilvafilho.petfunding.R
import com.squareup.picasso.Picasso

internal class ImagePagerAdapter(var mContext: Context, var mResources:List<String>) : PagerAdapter() {

    var mLayoutInflater: LayoutInflater

    init {
        mLayoutInflater = mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
    }

    override fun getCount(): Int {
        return mResources.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object` as LinearLayout
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val itemView = mLayoutInflater.inflate(R.layout.image_pager, container, false)

        val imageView = itemView.findViewById<ImageView>(R.id.imageView)
        Picasso.with(mContext)
                .load(mResources[position])
                .into(imageView)

        container.addView(itemView)

        return itemView
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        container.removeView(`object` as LinearLayout)
    }
}